File: 225. Implement Stack using Queues.py
class Stack(object):
    def __init__(self):
        """
        initialize your data structure here.
        """
        self.q1 = []
        self.q2 = []
        
    def push(self, x):
        """
        :type x: int
        :rtype: nothing
        """
        self.q1 += [x]

    def pop(self):
        """
        :rtype: nothing
        """
        while len(self.q1) > 1:
            self.q2 += [self.q1.pop(0)]
        self.q1.pop(0)
        while len(self.q2) > 0:
            self.q1 += [self.q2.pop(0)]

    def top(self):
        """
        :rtype: int
        """
        while len(self.q1) > 1:
            self.q2 += [self.q1.pop(0)]
        temp = self.q1[0]
        self.q2 += [self.q1.pop(0)]
        while len(self.q2) > 0:
            self.q1 += [self.q2.pop(0)]
        return temp

    def empty(self):
        """
        :rtype: bool
        """
        if len(self.q1) == 0:
            return True
        return False

File: 225. Implement Stack using Queues.cpp
#include <queue>
class Stack {
private:
    queue <int> q1;
    queue <int> q2;
public:
    // Push element x onto stack.
    void push(int x) {
        q1.push(x);
    }

    // Removes the element on top of the stack.
    void pop() {
        while (q1.size() > 1)
        {
            q2.push(q1.front());
            q1.pop();
        }
        q1.pop();
        while (q2.size() > 0)
        {
            q1.push(q2.front());
            q2.pop();
        }
    }

    // Get the top element.
    int top() {
        while (q1.size() > 1)
        {
            q2.push(q1.front());
            q1.pop();
        }
        int temp = q1.front();
        q2.push(q1.front());
        q1.pop();
        while (q2.size() > 0)
        {
            q1.push(q2.front());
            q2.pop();
        }
        return temp;
    }

    // Return whether the stack is empty.
    bool empty() {
        return q1.empty();
    }
};
